airflow/providers/salesforce/sensors/tableau_job_status.py

```diff
@@ -0,0 +1,63 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Optional
+
+from airflow.providers.salesforce.hooks.tableau import TableauHook
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class TableauJobStatusSensor(BaseSensorOperator):
+    """
+    Watches the status of a Tableau Server Job.
+
+    .. seealso:: https://tableau.github.io/server-client-python/docs/api-ref#jobs
+
+    :param job_id: The job to watch.
+    :type job_id: str
+    :param site_id: The id of the site where the workbook belongs to.
+    :type site_id: Optional[str]
+    :param tableau_conn_id: The Tableau Connection id containing the credentials
+        to authenticate to the Tableau Server.
+    :type tableau_conn_id: str
+    """
+
+    template_fields = ('job_id',)
+
+    @apply_defaults
+    def __init__(self,
+                 job_id: str,
+                 *args,
+                 site_id: Optional[str] = None,
+                 tableau_conn_id: str = 'tableau_default',
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.tableau_conn_id = tableau_conn_id
+        self.job_id = job_id
+        self.site_id = site_id
+
+    def poke(self, context):
```

Comment:
```suggestion
    def poke(self, context: dict) -> bool:
```
Type annotations ftw

Reply:
Yes 👍 Thanks.